api/src/main/java/org/apache/iceberg/Scan.java

```diff
@@ -195,4 +195,15 @@ default ThisT metricsReporter(MetricsReporter reporter) {
     throw new UnsupportedOperationException(
         this.getClass().getName() + " doesn't implement metricsReporter");
   }
+
+  /**
+   * Create a new scan that returns files with at least the given number of rows.
+   *
+   * @param numRows The minimum number of rows requested
```

Comment:
IMHO we would need to define this like we did in spec ? like its a hint and scan is strictly not required to return that many rows ?